kernel/kernel-api/src/test/scala/io/delta/kernel/internal/PaginationFilteredColumnarBatchSuite.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.util.Optional
+
+import io.delta.kernel.data.{ColumnarBatch, ColumnVector}
+import io.delta.kernel.internal.data.PaginationFilteredColumnarBatch
+import io.delta.kernel.test.VectorTestUtils
+import io.delta.kernel.types.{LongType, StructField, StructType}
+
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+
+class PaginationFilteredColumnarBatchSuite extends AnyFunSuite with VectorTestUtils with Matchers {
+
+  test("constructor should succeed when selectionVector is present and numSelectedRows is valid") {
+    val data = createTestColumnarBatch(5)
+    val selectionVector = Optional.of(booleanVector(Seq(true, false, true, false, true)))
+    val batch = new PaginationFilteredColumnarBatch(data, selectionVector, "/test/path", 3)
+
+    batch.getFilePath shouldBe "/test/path"
+    batch.getPreComputedNumSelectedRows shouldBe 3
+    batch.getData shouldBe data
+    batch.getSelectionVector shouldBe selectionVector
+  }
+
+  test(
+    "constructor should succeed when " +
+      "selectionVector is empty and numSelectedRows equals batch size") {
+    val data = createTestColumnarBatch(5)
+    val selectionVector = Optional.empty[ColumnVector]()
+    val batch = new PaginationFilteredColumnarBatch(data, selectionVector, "/test/path", 5)
+
+    batch.getFilePath shouldBe "/test/path"
+    batch.getPreComputedNumSelectedRows shouldBe 5
+    batch.getData shouldBe data
+    batch.getSelectionVector shouldBe selectionVector
+  }
+
+  test("constructor should throw IllegalArgumentException " +
+    "when selectionVector is empty and numSelectedRows != batch size") {
+    val data = createTestColumnarBatch(5)
+    val selectionVector = Optional.empty[ColumnVector]()
+    val exception = intercept[IllegalArgumentException] {
+      new PaginationFilteredColumnarBatch(data, selectionVector, "/test/path", 3)
+    }
+
+    exception.getMessage should include("Invalid precomputedNumSelectedRows")
+    exception.getMessage should include(
```

Comment:
tip: you can do `val exMsg = intercept[XXX} { stuff }.getMessage` to make these lines shorter and fit on one line